packages/regl-worldview/src/utils/HitmapObjectIdManager.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+
+//  Copyright (c) 2018-present, GM Cruise LLC
+//
+//  This source code is licensed under the Apache License, Version 2.0,
+//  found in the LICENSE file in the root directory of this source tree.
+//  You may not use this file except in compliance with the License.
+
+import last from "lodash/last";
+
+import Command from "../commands/Command";
+import type { ObjectHitmapId, Vec4, MouseEventObject } from "../types";
+import { intToRGB } from "./commandUtils";
+
+function fillArray(start: number, length: number): number[] {
+  return new Array(length).fill(0).map((_, index) => start + index);
+}
+
+/*
+ * This object manages the mapping between objects that are rendered into the scene and their IDs.
+ * It supplies an API for generating IDs for a rendered object and then accessing those objects based on their ID.
+ */
+export default class HitmapObjectIdManager {
+  _objectsByObjectHitmapIdMap: { [ObjectHitmapId]: Object } = {};
+  _objectToCommandMap: Map<Object, Command> = new Map();
```

Comment:
naming nit: you have `_foosByBarMap` above, and `_barToFooMap` here. Should change these to be consistent — personally I'd advocate for just `_foosByBar`, but it's up to you, just want to see them be consistent